### PR TITLE
Implement new dashboard widget interfaces

### LINKED ISSUE #4032: Implement IButtonWidget and IIconWidget interfaces
They are brought by https://github.com/nextcloud/server/pull/33658

### LINKED ISSUE #4034: Adapt upcoming cards dashboard widget for mobile clients
### Is your feature request related to a problem? Please describe.

With https://github.com/nextcloud/server/pull/33658 mobile clients can access app widget data.

### Describe the solution you'd like

Adapt the *upcoming cards* widget to https://github.com/nextcloud/server/pull/33658 once it's done.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by eneiluj**
requires https://github.com/nextcloud/server/pull/33658
closes #4032 
closes #4034 

Implement IAPIWidget, IButtonWidget and IIconWidget to provide information to clients.

Also make a small change in the dashboard widget component. It can be removed it if it's too much out of scope.

### CODE REVIEW

File: lib/Dashboard/DeckWidget.php [hunk 1]
```diff
@@ -68,17 +85,95 @@ public function getIconClass(): string {
...
+				WidgetButton::TYPE_NEW,
+				$this->urlGenerator->getAbsoluteURL(
+					$this->urlGenerator->linkToRoute(Application::APP_ID . '.page.index')
+				),
+				$this->l10n->t('Add card')
+			),
+			new WidgetButton(
+				WidgetButton::TYPE_MORE,
+				$this->urlGenerator->getAbsoluteURL(
+					$this->urlGenerator->linkToRoute(Application::APP_ID . '.page.index')
```
Comment (tobiasKaminsky) [inline lib/Dashboard/DeckWidget.php:173]: URL of "new" and "more" is the same?
Not sure if we can have another link or if we then want to omit one?

File: lib/Dashboard/DeckWidget.php [hunk 2]
```diff
@@ -68,17 +85,95 @@ public function getIconClass(): string {
...
+				),
+				$this->l10n->t('Add card')
```
Comment (tobiasKaminsky) [inline lib/Dashboard/DeckWidget.php:175]: On more button it shall be "Load more"

Comment (eneiluj) [reply lib/Dashboard/DeckWidget.php]: :+1: It's now "More cards" as it points to a page. I feel like "Load more" can lead to think it's going to load them in the client but it's just a link :grin:.

File: lib/Dashboard/DeckWidget.php [hunk 3]
```diff
@@ -68,17 +85,88 @@ public function getIconClass(): string {
...
+	 * @inheritDoc
+	 */
+	public function getWidgetButtons(string $userId): array {
+		return [
+			new WidgetButton(
+				WidgetButton::TYPE_MORE,
+				$this->urlGenerator->getAbsoluteURL(
+					$this->urlGenerator->linkToRoute(Application::APP_ID . '.page.index')
+				),
+				$this->l10n->t('More cards')
```
Comment (tobiasKaminsky) [inline lib/Dashboard/DeckWidget.php:168]: AFAIK Jan wants to have always "load more", only.

Comment (eneiluj) [reply lib/Dashboard/DeckWidget.php:168]: Even if it does not load anything but just opens a browser?

Comment (tobiasKaminsky) [reply lib/Dashboard/DeckWidget.php:168]: @jancborchardt ^

Comment (eneiluj) [reply lib/Dashboard/DeckWidget.php]: @tobiasKaminsky It's now "Load more". We can change that later if needed.

**Comment by eneiluj** [comment]
Rebased on master.
We can ignore the Psalm complaints as I don't know how to remove them but can assure it works :grin:.

**Review by tobiasKaminsky — APPROVED** [review]
Tested and works from client side :+1:

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**